repository/documents.go

```diff
@@ -0,0 +1,38 @@
+package repository
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+)
+
+func isMetaURI(u string) bool {
+	return strings.HasPrefix(u, "system://") &&
+		strings.HasSuffix(u, "/meta")
+}
```

Comment:
Yeah, but that's by design. If it quacks like a meta URI we will treat it as a meta document (with the constraints and rules that comes from that), but it will also be parsed down the line. But yeah, it suffers from the same thing that you noted below.

Reply:
It's not a public function so it's no big deal really :)